client-lib/java/src/main/java/com/abcxyz/jvs/JustificationDeserializer.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.abcxyz.jvs;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.deser.std.StdDeserializer;
+import java.io.IOException;
+
+public class JustificationDeserializer extends StdDeserializer<Justification> {
```

Comment:
Not clear to me why we're creating a custom deserializer rather than the standard object mapper. 

```
ObjectMapper objectMapper = new ObjectMapper();
Justification justification = objectMapper.readValue(json, Justification.class);
```

Reply:
Two reasons:

1. I didn't know about `ObjectMapper`. The existing implementation manually creates a HashMap and isn't tied to structured values.
2. I prefer an interim data structure in case we need to mutate fields in the future for backwards compatability.

Reply:
ObjectMapper is the standard solution in Java. I'm hesitant to go with a custom deserializer when we don't have a strong reason to have one.

Reply:
I guess I'm confused why we didn't use `ObjectMapper` in the first place then?

Reply:
Looks like we were converting it to a map, which already had a built in deserializer. ObjectMapper gets you custom objects easily, which we're doing now.

Reply:
Chatted with Ryan offline. The JWT library makes using ObjectMapper prohibitively difficult and is less readable.

Reply:
Talked with @sethvargo offline. It appears there is no easy way to get the string json representation of the "justs" claim to convert using the Object Mapper. Rather than use convoluted code to serialize to a string just to deserialize to Justifications, we can stick with this. The major thing i dislike is that now both the object and the deserializer have to be modified if there are new values.

I'm now wondering if it makes sense to except (or at least, log to keep backwards compatibility) on unhandled values